### PR TITLE
Added nuxt-auth for basic login

### PR DESCRIPTION
**Pull request by CapitaineToinon**
## What

Added nuxt-auth to be able to register, login and logout with the backend

## Link to Issue

https://github.com/leaderboardsgg/leaderboard-site/issues/103

### CODE REVIEW

**Comment by zysim** [comment]
Works like a dream, and all tests pass :sunglasses:. A few things though:
1. Could we get some tests for the `<SiteNavbar>`, to check which of the login/out and signin buttons render, depending on the situation?
2. We'd have to update the Readme too to talk about the `.env` file

File: package.json [hunk 1]
```diff
@@ -32,6 +34,7 @@
     "@testing-library/vue": "^5.8.2",
     "@vueuse/core": "^5.2.0",
     "core-js": "^3.16.0",
+    "dotenv": "^10.0.0",
```
Comment (exodustx0) [inline package.json:37]: I _strongly_ urge you to use [`dotenv-safe`](https://www.npmjs.com/package/dotenv-safe) (and with that, `@types/dotenv-safe`) instead: it uses `dotenv` under the hood, but with the addition that it throws if `.env` and `.env.example` don't have the exact same fields contained in them. This prevents any env-related surprises down the line. It _does_ use an older version of `dotenv` however (8.2.0, as opposed to 10.0.0), but I don't expect that to be a dealbreaker, looking at the [changelog](https://github.com/motdotla/dotenv/blob/master/CHANGELOG.md).

**Review by exodustx0 — CHANGES_REQUESTED** [review]

**Comment by legrandgrand** [comment]
TODO (maybe in other PRs?):
* Error handling: When there is an error, show what the error is and don't redirect to 400 error page
* Form validation
* Confirmation messages

**Comment by legrandgrand** [comment]
This PR doesn't fix EVERYTHING listed in TODOs: Error Handling in form and  Form validation will be done in another issue; Confirmation and Error messages globally will be handled through snackbars in another Issue as well.

The aim is to have a backbone for API calls to backend with this feature, which will allow other issues to be created and solved.

**Review by erunks — APPROVED** [review]
Looks good to me

**Comment by zysim** [comment]
Code looks good, and I can confirm the flow works still :sunglasses:

**Pull request merged**